src/librvnpal/src/posix/journal.c

```diff
@@ -0,0 +1,195 @@
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif 
+
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <stdio.h>
+#include <assert.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "rvn.h"
+#include "status_codes.h"
+#include "internal_posix.h"
+
+struct journal_handle
+{
+    int fd;
+    const char *path;
+};
+
+PRIVATE void free_journal_handle(struct journal_handle* handle)
```

Comment:
```suggestion
PRIVATE void _free_journal_handle(struct journal_handle* handle)
```

Because it is internal, no?